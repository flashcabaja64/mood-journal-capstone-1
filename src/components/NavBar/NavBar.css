li {
  display: inline-block;
}

.nav-link {
  display: block;
  color: white;
  text-align: center;
  padding: 1.5rem 1.5rem;
  text-decoration: none;
}

.nav-link:hover {
  background-color:#0E182D;
}

.app-title {
  text-decoration: none;
  display: inline-block;
  color: white;
  padding: 14px 16px;
  font-family: 'Chango', cursive;
  margin: 0 auto;
  font-size: 2rem;
}

nav.nav {
  background-color: #1A5486;
}
