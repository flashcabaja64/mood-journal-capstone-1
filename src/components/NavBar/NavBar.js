import React, { Component, Fragment } from 'react';
import { NavLink } from 'react-router-dom';
import TokenService from '../../services/token-service'
import NavContext from '../../MoodContext/NavContext'
//import MoodPage from '../../pages/MoodPage/MoodPage'
import './NavBar.css'

//refer to defined routes in App.js, be consistent on the route naming.
//create log-out link
export default class NavBar extends Component {

  static contextType = NavContext

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    TokenService.clearUserId();
    this.context.setAuthToken(TokenService.getAuthToken())
  }

  renderUserName = () => {
    if(this.context.authToken) {
      return <h3 className='username'>{TokenService.getUserName()}</h3>
    } else {
      return <h3 className='username'>Guest</h3>
    }
  }

  renderLogoutLink() {
    return (
      <nav className="log-in">
        {this.renderUserName}
        <li>
          <NavLink
            to='/home'
            activeClassName="active"
            className="nav-link">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink
            to='/moods'
            activeClassName="active"
            className="nav-link">
            My Moods
          </NavLink>
        </li>
        <li>
          <NavLink
            to='/newmoods'
            activeClassName="active"
            className="nav-link">
            New Mood
          </NavLink>
        </li>
        <li>
          <NavLink
            to='/dashboard'
            activeClassName="active"
            className="nav-link">
            Dashboard
          </NavLink>
        </li>
        <li>
          <NavLink
            className="nav-link"
            onClick={this.handleLogoutClick}
            exact
            to='/home'>
            Logout
          </NavLink>
        </li>
      </nav>
    )
  }

  renderLoginLink() {
    return (
      <Fragment>
        <li>
          <NavLink
            to='/home'
            activeClassName="active"
            className="nav-link">
            Home
          </NavLink>
        </li>
        <li>
        <NavLink
          to="/register"
          activeClassName="active"
          className="nav-link">
          Register
        </NavLink>
        </li>
        <li>
        <NavLink
          to='/login'
          className="nav-link">
          Log-in
        </NavLink>
        </li>
      </Fragment>
    )
  }

  render() {
    return (
      <nav className='nav'>
        <header className="main-app-name">
        <NavLink
          exact
          to="/"
          className="app-title">
          Mood Journal
        </NavLink>
          <Fragment>
            {this.context.authToken
              ? this.renderLogoutLink()
              : this.renderLoginLink()}
          </Fragment>
        </header>
      </nav>
    )
  }
}