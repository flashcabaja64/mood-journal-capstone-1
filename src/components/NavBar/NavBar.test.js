import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from './NavBar'
import {MemoryRouter} from 'react-router';
import renderer from 'react-test-renderer';

describe('NavBar component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<MemoryRouter><NavBar /></MemoryRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><NavBar /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

})
