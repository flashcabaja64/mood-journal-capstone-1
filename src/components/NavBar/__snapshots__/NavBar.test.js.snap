// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component testing renders UI as expected 1`] = `
<nav
  className="nav"
>
  <header
    className="header"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      <h2
        className="app-title"
      >
        Mood Journal
      </h2>
    </a>
  </header>
  <li>
    <a
      aria-current={null}
      className="nav-link"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="nav-link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="nav-link"
      href="/login"
      onClick={[Function]}
    >
      Log-in
    </a>
  </li>
</nav>
`;
